node/common/client_service_base.ts

```diff
@@ -291,10 +301,257 @@ export class ClientServiceBase {
+              ContractExecutionResult.fromGrpcContractExecutionResponse(
+                response as ContractExecutionResponse
+              )
+            );
+          }
+
+          try {
+            const auditorResponse = await this.validateExecution(
+              ordered,
+              response as ContractExecutionResponse
+            );
+
+            const isConsistent = this.validateResponses(
+              response as ContractExecutionResponse,
+              auditorResponse
+            );
+
+            if (!isConsistent) {
+              return reject(
+                new ClientError(
+                  StatusCode.INCONSISTENT_STATES,
+                  "The results from Ledger and Auditor don't match"
+                )
+              );
+            }
+
+            return resolve(
+              new ContractExecutionResult(
+                (response as ContractExecutionResponse).getContractResult(),
+                (response as ContractExecutionResponse).getFunctionResult(),
+                (response as ContractExecutionResponse)
+                  .getProofsList()
+                  .map(p => AssetProof.fromGrpcAssetProof(p)),
+                auditorResponse
+                  .getProofsList()
+                  .map(p => AssetProof.fromGrpcAssetProof(p))
+              )
+            );
+          } catch (err) {
+            return reject(err);
+          }
+        }
+      );
+    });
+
+    return this.executePromise(promise) as Promise<ContractExecutionResult>;
+  }
+
+  /**
+   * Create the byte array of ContractExecutionRequest
+   * @param {string} contractId
+   * @param {Object|string} contractArgument
+   * @param {string|null} [functionId=null]
+   * @param {Object|string|null} [functionArgument=null]
+   * @param {string|null} [nonce=null]
+   * @return {Promise<Uint8Array>}
+   * @throws {Error}
+   */
+  async createSerializedExecutionRequest(
+    contractId: string,
+    contractArgument: Object | string,
+    functionId: string | null = null,
+    functionArgument: Object | string | null = null,
+    nonce: string | null = null
+  ): Promise<Uint8Array> {
+    return (
+      await this.createContractExecutionRequest(
+        contractId,
+        functionId,
+        contractArgument,
+        functionArgument,
+        nonce
+      )
+    ).serializeBinary();
+  }
+
+  /**
+   * Create the byte array of ContractExecutionRequest
+   * @deprecated Use {@link createSerializedExecutionRequest} instead
+   * @param {string} contractId
+   * @param {Object} argument
+   * @param {Object|null} [functionArgument=null]
+   * @return {Promise<Uint8Array>}
+   * @throws {Error}
+   */
+  async createSerializedContractExecutionRequest(
+    contractId: string,
+    argument: Object,
+    functionArgument: Object | null = null
+  ): Promise<Uint8Array> {
+    return this.createSerializedExecutionRequest(
+      contractId,
+      argument,
+      '',
+      functionArgument,
+      uuidv4()
```

Comment:
There is another place to generate an UUID in `createContractExecutionRequest()`. Can we generate it only there for maintainability and readability?

Reply:
Ah! Thank you.
Let's remove this redundancy.

I removed it in https://github.com/scalar-labs/scalardl-javascript-client-sdk/pull/22/commits/41b163a9b0d649265ee46c6366c97f3b4d2a525d